### DISCUSSION TITLE
Accessing acceleration data

### DISCUSSION THREAD

**Discussion by vannem95** [USC-ACTLab/crazyswarm#649] [Q&A] [answered]
What is the best way to access crazyflie drone acceleration data in a python script?

**Comment by jpreiss** [comment, marked as answer]
Motion capture systems don't measure velocity or acceleration. You can log the Crazyflie's onboard state estimate with the logging system. Be aware of latency if you are using it for feedback control.

Estimating derivatives with finite differences of `position()` values might also work, but you need to be careful about noise (and also remember that lowpass filtering adds delay).

**Reply by vannem95** [reply]
Hi James. Thanks for the quick response. 
Is there a simple method to access the logging system through crazyswarm or do i just have to go thru the crazyflie documentation and figure it out?
Also, I have been using finite differencing so far. If you did this in the past, how did you manage the filter delay issue?
Appreciate your help.

**Reply by knmcguire** [reply]
For the logging documentation, we rendered all the logging possibilities here on the bitcraze website: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/

**Reply by vannem95** [reply]
Thanks, @knmcguire . I got the logging to work but i keep running into latency issues. Are there any ways to minimize it?

**Reply by knmcguire** [reply]
Ah no sorry, that would be something that is within the crazyswarm server, so I wouldn't know how to do that... have you tried to compare it with the [python libary cflib](https://github.com/bitcraze/crazyflie-lib-python)? There shouldn't really be a difference I thought, but at least that is an more direct way of getting it directly in the python script as you mentioned

**Reply by vannem95** [reply]
hi @jpreiss and @knmcguire:
So, i tried logging by adding this 
`genericLogTopic_log1_Variables: ["acc.x", "acc.y"]` to the launch file and
`rospy.Subscriber("/cf1/log1", GenericLogData, callback)` to the python script

It worked a couple of times and after that i kept running into latency issues like this:
`[ WARN] [1653496554.671877987]: Latency too high! Is 0.009569 s.`
`[ WARN] [1653496571.504166403]: Latency too high! Is 0.010865 s.`
`[ INFO] [1653496610.430896314]: [cf1] Takeoff`
`[ INFO] [1653496610.443354510]: [cf1] Update parameters`
`[ INFO] [1653496615.022663440]: [cf1] WARNING: LOG packets drop dete01 packets lost)`
`terminate called after throwing an instance of 'std::runtime_error'`
`  what():  Unknown rigid body!`
`[crazyswarm_server-2] process has died [pid 839129, exit code -6, cmd /home/david/crazyswarm/ros_ws/devel/lib/crazyswarm/crazyswarm_server __name:=crazyswarm_server __log:=/home/david/.ros/log/bd08f810-dc48-11ec-ba4b-f7911a0afd79/crazyswarm_server-2.log].`
`log file: /home/david/.ros/log/bd08f810-dc48-11ec-ba4b-f7911a0afd79/crazyswarm_server-2*.log`

And now I can't even get 'hello_world.py' to work as I get the same latency errors and the drone would go crazy. The drone marker on rviz moves multiple seconds after the drone crashes. 

My guess is this issue has something to do with the memory/cache of the drone. After I started logging the memory/cache got full and it's causing latency everywhere even with 'hello_world.py'

What do you guys think?

**Reply by jpreiss** [reply]
It's normal to get a few latency warnings at startup. It's only a problem if they happen continuously.

The "unknown rigid body" and the high latency of mocap in rviz suggest that there is something wrong with your mocap setup.

**Reply by vannem95** [reply]
Hi @jpreiss . I appreciate the quick response. I'm getting the latency errors almost all the time. But they seem to have started when I started doing logging. Is it possible to clear the drone log memory/cache? Or are there any other solutions to latency issues?

I will try resetting/troubleshooting my vicon mocap system.

**Reply by jpreiss** [reply]
Try reducing the log frequency?

**Reply by vannem95** [reply]
I'll try that. Thank you.
